Python/Product/PythonTools/PythonTools/Debugger/DebugAdapter/DebugAdapterLauncher.cs

```diff
@@ -185,29 +202,26 @@ public sealed class DebugAdapterLauncher : IAdapterLauncher {
         }
 
         [DllImport("shell32.dll")]
-        static extern IntPtr CommandLineToArgvW([MarshalAs(UnmanagedType.LPWStr)] string lpCmdLine, out int pNumArgs);
+        private static extern IntPtr CommandLineToArgvW([MarshalAs(UnmanagedType.LPWStr)] string lpCmdLine, out int pNumArgs);
 
         private static IEnumerable<string> GetParsedCommandLineArguments(string command) {
-            if (String.IsNullOrEmpty(command)) {
+            if (string.IsNullOrEmpty(command)) {
                 yield break;
             }
 
-            IntPtr argPointer = CommandLineToArgvW(command, out int argumentCount);
-            string[] arguments = new string[argumentCount];
-
-            if (argPointer != IntPtr.Zero) {
-                try {
-                    for (int i = 0; i < arguments.Length; i++) {
-                        yield return Marshal.PtrToStringUni(Marshal.ReadIntPtr(argPointer, i * IntPtr.Size));
-                    }
+            IntPtr argPointer = CommandLineToArgvW(command, out var argumentCount);
+            if (argPointer == IntPtr.Zero) {
+                throw new System.ComponentModel.Win32Exception();
```

Comment:
If the intent is to capture the `GetLastError()` code from the failed call, then the P/Invoke declaration [needs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getlastwin32error?view=netframework-4.8#remarks) `[DllImport(..., SetLastError=true)]`.

Reply:
Don't think VS needs to display the exact error to the user--just state the fact that VS can't parse. Though, the error code could be useful for debugging.

Reply:
I was looking for `catch (Win32Exception)`, and didn't find it - now I see that it catches `Exception`.

The parameterless `Win32Exception()` constructor will try to retrieve the error code, and then use `FormatMessage` to turn it into a string. So it will be whatever was the last P/Invoke with `SetLastError=true`. If it's not the one above, it will be something else (quite possibly, something called from within .NET stdlib). If it's not actually displayed, it's not a big deal, but it may be confusing when debugging, because you'll see the exception happen - it'll get logged in Output, for example - but its message will be unrelated to the actual error.

Reply:
Good point. Will be updated in next commit.